docs/DeveloperGuide.md

```diff
@@ -6,33 +6,230 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### The Overall UML Diagram:
 
+![UML Diagram](assets/images/mathGenius.png)    
+
+
+### The overall Sequence Diagram:
+![Sequence Diagram](assets/images/sequenceDiagram.png)    
+
+
+
+### problemGenerator Component
+API: [ProblemGenerator.java](../src/main/java/seedu/duke/ProblemGenerator.java)
+
```

Comment:
Are these links functional? Perhaps you could provide functional links to the code or omit them if not necessary.
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/108778858/83eb12ef-88f8-4253-a451-d6a29eedcdd6)